diff --git a/configure b/configure
index e39d1c2..d96f251 100755
--- a/configure
+++ b/configure
@@ -8207,9 +8207,9 @@ printf "%s\n" "$as_me: Use gpgrt-config with $gpgrt_libdir as gpg-error-config"
     req_minor=`echo $min_gpg_error_version | \
                sed 's/\([0-9]*\)\.\([0-9]*\)/\2/'`
     major=`echo $gpg_error_config_version | \
-               sed 's/\([0-9]*\)\.\([0-9]*\).*/\1/'`
+               sed 's/\([0-9]*\)\.\([0-9]*\).*/\1/' | cut -f 3 -d' '`
     minor=`echo $gpg_error_config_version | \
-               sed 's/\([0-9]*\)\.\([0-9]*\).*/\2/'`
+               sed 's/\([0-9]*\)\.\([0-9]*\).*/\2/' | cut -f 3 -d' '`
     if test "$major" -gt "$req_major"; then
         ok=yes
     else
@@ -14502,11 +14502,11 @@ else
       if test "$pinentry_gnome_3" = "yes"; then
         PINENTRY_DEFAULT=pinentry-gnome3
       else
-        if test "$pinentry_curses" = "yes"; then
-          PINENTRY_DEFAULT=pinentry-curses
+        if test "$pinentry_tty" = "yes"; then
+          PINENTRY_DEFAULT=pinentry-tty
         else
-          if test "$pinentry_tty" = "yes"; then
-            PINENTRY_DEFAULT=pinentry-tty
+          if test "$pinentry_curses" = "yes"; then
+            PINENTRY_DEFAULT=pinentry-curses
           else
             if test "$pinentry_w32" = "yes"; then
               PINENTRY_DEFAULT=pinentry-w32
